pkg/ccl/changefeedccl/scheduled_changefeed_test.go

```diff
@@ -0,0 +1,677 @@
+	"github.com/cockroachdb/cockroach/pkg/jobs"
+	"github.com/cockroachdb/cockroach/pkg/jobs/jobspb"
+	"github.com/cockroachdb/cockroach/pkg/jobs/jobstest"
+	"github.com/cockroachdb/cockroach/pkg/scheduledjobs"
+	"github.com/cockroachdb/cockroach/pkg/scheduledjobs/schedulebase"
+	"github.com/cockroachdb/cockroach/pkg/security/username"
+	"github.com/cockroachdb/cockroach/pkg/sql/parser"
+	"github.com/cockroachdb/cockroach/pkg/sql/sem/tree"
+	"github.com/cockroachdb/cockroach/pkg/sql/sessiondata"
+	"github.com/cockroachdb/cockroach/pkg/testutils"
+	"github.com/cockroachdb/cockroach/pkg/testutils/serverutils"
+	"github.com/cockroachdb/cockroach/pkg/testutils/sqlutils"
+	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
+	"github.com/cockroachdb/cockroach/pkg/util/log"
+	"github.com/cockroachdb/cockroach/pkg/util/timeutil"
+	"github.com/cockroachdb/errors"
+	pbtypes "github.com/gogo/protobuf/types"
+	"github.com/stretchr/testify/require"
+)
+
+const allSchedules = 0
+
+// testHelper starts a server, and arranges for job scheduling daemon to
+// use jobstest.JobSchedulerTestEnv.
+// This helper also arranges for the manual override of scheduling logic
+// via executeSchedules callback.
+type execSchedulesFn = func(ctx context.Context, maxSchedules int64) error
+type testHelper struct {
+	env              *jobstest.JobSchedulerTestEnv
+	cfg              *scheduledjobs.JobExecutionConfig
+	sqlDB            *sqlutils.SQLRunner
+	server           serverutils.TestServerInterface
+	executeSchedules func() error
+	createdSchedules []int64
+}
+
+func newTestHelper(t *testing.T) (*testHelper, func()) {
+	sh := &testHelper{
+		env: jobstest.NewJobSchedulerTestEnv(
+			jobstest.UseSystemTables, timeutil.Now(), tree.ScheduledChangefeedExecutor),
+	}
+
+	s, db, stopServer := startTestFullServer(t, makeOptions(withSchedulerHelper(sh)))
+	sh.sqlDB = sqlutils.MakeSQLRunner(db)
+	sh.server = s
+	return sh, stopServer
+}
+
+// withScheduleHelper returns an option to configure test server with
+// settings required to make scheduled jobs system function in tests.
+func withSchedulerHelper(sh *testHelper) func(opts *feedTestOptions) {
+	return func(opts *feedTestOptions) {
+		opts.knobsFn = func(knobs *base.TestingKnobs) {
+			jobsKnobs := knobs.JobsTestingKnobs.(*jobs.TestingKnobs)
+			jobsKnobs.JobSchedulerEnv = sh.env
+			jobsKnobs.TakeOverJobsScheduling = func(fn execSchedulesFn) {
+				sh.executeSchedules = func() error {
+					defer sh.server.JobRegistry().(*jobs.Registry).TestingNudgeAdoptionQueue()
+					return fn(context.Background(), allSchedules)
+					//return sh.cfg.DB.Txn(context.Background(), func(ctx context.Context, txn *kv.Txn) error {
+					//	return fn(ctx, allSchedules)
+					//})
+				}
+			}
+			jobsKnobs.CaptureJobExecutionConfig = func(config *scheduledjobs.JobExecutionConfig) {
+				sh.cfg = config
+			}
+			// Revisit: we no longer have support for asof. so we can remove this. we no longer need
```

Comment:
just delete the code that's not used.
ASOF is used in backup code; I suspect that we will have the use for ASOF clause in the future when we support incremental exports.  FOr now, just drop this.